frontend/packages/console-dynamic-plugin-sdk/src/app/modal-support/ModalProvider.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+
+type CloseModal = () => void;
+
+export type ModalComponent = () => React.FC<{
```

Comment:
We store the value in a state. When calling setState it executes the function if a function is passed as a value to it. To avoid that I have set the signature as above. I will update it to React.FC directly and handle the `setState` issue internally.